app/models/book.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class Book < ApplicationRecord
+  self.inheritance_column = nil
+
+  # enums
+  enum activity: { deleted: 0, active: 1 }
+  enum contribution_rate: { all_of: 0, chapters: 1 }
+  enum scope: { national: 0, international: 1 }
+  enum type: {
+    scientific:           25,
+    textbook:             26,
+    encyclopedia_article: 28,
+    translation:          29,
+    research:             43
+  }
```

Comment:
Buradan anladığım kadarıyla model isminin "**Publication**" olması daha uygun olacağı görüşündeyim. **Yayınlar** içinde **tip** olarak tanımlananları karşılamakta. "Book" çok spesifik bir isim.

Reply:
Hocam, makale ve bildiriler de bir publication aslında. Sadece kitap için bunu kullanırsak karışıklık olmaz mı?

Reply:
- Daha dikkatli baktığımda "**Article**" modelini de gördüm. STI (Single-Table Inheritance) bunun ideal çözümü olarak görünüyor. Hem ilişkilerin takibinin kolaylığı hem de gerekli rutinlerin bir arada toplanmasını sağlaması nedeniyle. "**Publication**" class'ından türeyecek **Book**, **Article**, **Conference**, vb. modellerinden oluşacak. Böylece hem her nesneyi ayrı ayrı kontrol ediyor olabileceksin hem de temel rutinler tek noktada toplanarak kod tekrarının önüne geçmiş olacaksın.

- Senin gerçeklediğin çözümü kullanabilmemiz için şayet **type** tanımı **YÖKSİS bağımlı bir veri değilse** bunu *type* tanımında çözmek en uygunu olacaktır:
```
  enum type: {
    scientific:           25,
    textbook:             26,
    encyclopedia_article: 28,
    translation:          29,
    research:             43,
    article:              XX,
    conference:           YY
  }
```

Ama bu şekilde çözmüş olsak dahi model isminin **Publication** olarak değişmesi gerekmekte.

Reply:
Hocam, Yöksis'te kitap, makale ve bildirilerin kendine ait `type` alanları var. STI kullanamayacak kadar da farklı alanlara sahipler.
İlla Book model ismini kullanalım demiyorum. İleride öngöremediğim herhangi bir süreçte karışıklığa sebebiyet verecekse elbette başka bir isimlendirme kullanırsak iyi olur.